src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    }
+
+    if (action.type === 'addProperties') {
+      currentState = {
+        ...currentState,
+        ...action.extraData,
+      };
```

Comment:
Could use `Object.assign` instead of spreading here (already creating a copy of the object before `.push`ing)